src/generator.rs

```diff
@@ -90,6 +91,31 @@ pub fn run_generator(devices: &[Device], output_directory: &Path, fake_mode: boo
                 < max_device
             {}
         }
+
+        let compressions: Vec<&String> = devices
+            .iter()
+            .filter(|device| !device.compression_algorithm.is_none())
+            .map(|device| device.compression_algorithm.as_ref().unwrap())
+            .collect::<HashSet<_>>()
+            .into_iter()
+            .collect();
+
+        if !compressions.is_empty() {
+            let known = get_known_compressions(None)?;
+
+            for compression in compressions.into_iter().filter(|c| !known.contains(c)) {
```

Comment:
If you make `get_known_compressions()` return a `BTreeSet` (since, well, that's what it represents), you'll be able to do `compressions.intersection(known)` here, which won't be quadratic.